Zip 321 address bundle extension (#455)
zcash/zips: zip-0321.rst

  zcashurn        = "zcash:" ( zcashaddress [ "?" zcashparams ] / "?" zcashparams )
  zcashaddress    = 1*( ALPHA / DIGIT )

>   zcashaddrbundle = zcashaddress [ ";" zcashaddrbundle ]

Reviewer: ```suggestion
  zcashaddrbundle = zcashaddress [ "+" zcashaddrbundle ]
```
I suggest making this change so that a bundle, converted to uppercase, falls within the [Alphanumeric subset](https://en.wikipedia.org/wiki/QR_code#Storage) of a QR code. The intersection of characters in that subset and in `allowed-delims` is `"$" / "*" / "+"`, of which `+` seems the most semantically appropriate. (`:` is also a possibility, but that's in `qchar` rather than `allowed-delims`.)

URI of the form ``zcash:?address=<address>&...`` where ``<address>`` is an
instance of ``zcashaddress``.

> If there are any non-address parameters having a given ``paramindex``, then the URI MUST
> contain an ``address`` parameter and/or an ``bundle`` parameter having that

Reviewer: ```suggestion
If there are any non-address parameters having a given ``paramindex``, then the URI 
MUST contain an ``address`` parameter and/or a ``bundle`` parameter having that
```

be expected that senders will have funds available in the Sprout pool with which
to satisfy requests for payment to a Sprout address.

An instance of ``zcashaddrbundle`` represents a bundle of addresses to which a payment may
be sent. By providing a bundle of addresses, the requestor can ensure that the payer has
been provided with an address that the payer's wallet can recognize, even if that wallet

> has not yet been updated to support newly released protocol versions.  Implementations
> SHOULD select the address from the bundle that is associated with the most recent Zcash
> protocol version that they support. Each such bundle of addresses MUST be associated with
> only a single recipient; for separate recipients, distinct payments within the request can
> be used instead. A bundle of addresses MUST NOT contain more than one address of each
> address type.

Reviewer 2: ```suggestion
has not yet been updated to support newly released protocol versions.  Implementations
MUST select the address from the bundle that is associated with the most recent Zcash
shielded protocol version that they support. If there is no currently supported shielded 
protocol version address supplied, then fallback to the transparent address, if supplied. 
Each such bundle of addresses MUST be associated with only a single recipient; for 
separate recipients, distinct payments within the request can be used instead. A bundle 
of addresses MUST NOT contain more than one address of each address type.
```

If a ``bundle`` parameter appears at a given ``paramindex`` and an ``address`` parameter
appears at the same index, the value of the ``address`` parameter at that index MUST also

> be a member of the bundle. It is RECOMMENDED that the value of such an ``address``

Reviewer: Perhaps say that a recipient MAY ignore the `address` param if there is a `bundle` param.

---------------

If an amount is provided, it MUST be specified in decimal ZEC. If a decimal fraction

> is present then a period ('.') MUST be used as the separating character to separate

Reviewer: ```suggestion
is present then a period (``.``) MUST be used as the separating character to separate
```

address
   Zcash address string (shielded or transparent)

bundle
   Zcash Address Bundle. An address bundle is an abstraction that 
   provides Zcash users the ability to express a clear intent on how they want 
   to receive a payment without the need to restrict senders either to transparent
   or a specific Shielded Pool (Sapling or Orchard) at the moment of generating
   a payment request URI. When presented a Bundle, a sender can send the requested
   amount to the `best pool` (newest shielded pool) or fall back to transparent
   for the case sender has no capability for sending to shielded. Wallets sending

>    to transparent address must ignore the memo field if available.

Reviewer: ```suggestion
   to transparent addresses MUST ignore the memo field if available.
```

address
   Zcash address string (shielded or transparent)

bundle

>    Zcash Address Bundle. An address bundle is an abstraction that provides Zcash users the
>    ability to express a clear intent on how they want to receive a payment without the
>    need to restrict senders either to transparent or a specific Shielded Pool (Sapling or
>    Orchard). When presented a bundle, a sender can send the requested amount to the "best
>    pool" (the newest shielded pool) or fall back to sending to a transparent address in the
>    case that the sender has no capability for sending to any of the shielded addresses
>    provided. Wallets sending to a transparent addresses MUST ignore the memo field.

Reviewer 2: ```suggestion
   Zcash Address Bundle. An address bundle is an abstraction that provides Zcash users the
   ability to express a clear intent on how they want to receive a payment without the
   need to restrict senders either to transparent or a specific Shielded Pool (Sapling or
   Orchard). When presented a bundle, a sender MUST send the requested amount to the "best
   pool" (the newest shielded pool) or fall back to sending to a transparent address in the
   case that the sender has no capability for sending to any of the shielded addresses
   provided. Wallets sending to a transparent addresses MUST ignore the memo field.
```

.. [#protocol-networks] `Zcash Protocol Specification, Version 2020.1.15. Section 3.11: Mainnet and Testnet <protocol/protocol.pdf#networks>`_
.. [#protocol-saplingbalance] `Zcash Protocol Specification, Version 2020.1.15. Section 4.12: Balance and Binding Signature (Sapling) <protocol/protocol.pdf#saplingbalance>`_
.. [#protocol-transparentaddrencoding] `Zcash Protocol Specification, Version 2020.1.15. Section 5.6.1: Transparent Addresses <protocol/protocol.pdf#transparentaddrencoding>`_

> .. [#protocol-saplingpaymentaddrencoding] `Zcash Protocol Specification, Version 2020.1.16. Section 5.6.3.1: Sapling Payment Addresses <protocol/protocol.pdf#saplingpaymentaddrencoding>`_
> .. [#protocol-orchardpaymentaddrencoding] `Zcash Protocol Specification, Version 2020.1.16. Section 5.6.4.1: Orchard Payment Addresses <protocol/protocol.pdf#orchardpaymentaddrencoding>`_

Reviewer: ```suggestion
.. [#protocol-saplingpaymentaddrencoding] `Zcash Protocol Specification, Version 2020.1.17. Section 5.6.3.1: Sapling Payment Addresses <protocol/protocol.pdf#saplingpaymentaddrencoding>`_
.. [#protocol-orchardpaymentaddrencoding] `Zcash Protocol Specification, Version 2020.1.17. Section 5.6.4.1: Orchard Payment Addresses <protocol/protocol.pdf#orchardpaymentaddrencoding>`_
```
